kubernetes/GKE/README.md

```diff
@@ -0,0 +1,520 @@
+
+To manage a Kubernetes cluster, you use [kubectl](https://kubernetes.io/docs/user-guide/kubectl/), the Kubernetes command-line client. If you use GKE Cloud Shell, `kubectl` is already installed. To install `kubectl` locally, use the [gcloud components install](https://kubernetes.io/docs/tasks/tools/install-kubectl/) command:
+
+```bash
+gcloud components install kubectl
+```
+
+To configure `kubectl` to connect to your Kubernetes cluster, use the [gcloud container clusters get-credentials](https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials) command. This command downloads credentials and configures the Kubernetes CLI to use them.
+
+```bash
+gcloud container clusters get-credentials torchserve --region us-west1 --project pytorch-tests-261423
+Fetching cluster endpoint and auth data.
+kubeconfig entry generated for torchserve.
+```
+
+#### 1.4 Install helm
+
+```bash
+curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+chmod 700 get_helm.sh
+./get_helm.sh
+```
+
+### 2 Deploy TorchServe on GKE
+
+#### 2.1 Download the github repository and enter the kubernetes directory
+
+```bash
+git clone https://github.com/pytorch/serve.git
+
+cd serve/kubernetes/GKE
+```
+
+#### 2.2 Install NVIDIA device plugin
+
+Before the GPUs in the nodes can be used, you must deploy a DaemonSet for the NVIDIA device plugin. This DaemonSet runs a pod on each node to provide the required drivers for the GPUs.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+daemonset.apps/nvidia-driver-installer created
+```
+
+```bash
+kubectl get nodes "-o=custom-columns=NAME:.metadata.name,MEMORY:.status.allocatable.memory,CPU:.status.allocatable.cpu,GPU:.status.allocatable.nvidia\.com/gpu"
+``` 
+should show something similar to:
+
+```bash
+NAME                                        MEMORY       CPU     GPU
+gke-torchserve-default-pool-aa9f7d99-ggc9   12698376Ki   3920m   1
+```
+
+#### 2.3 Create a storage disk
+
+A standard storage class is created with google compute disk. If multiple pods need concurrent access to the same storage volume, you need Google NFS. Create the storage disk named *nfs-disk* with the following command:
+
+```bash
+gcloud compute disks create --size=200GB --zone=us-west1-a nfs-disk
+
+NAME     ZONE        SIZE_GB  TYPE         STATUS
+nfs-disk  us-west1-a  200      pd-standard  READY
+
+New disks are unformatted. You must format and mount a disk before it
+can be used. You can find instructions on how to do this at:
+
+https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
+
+```
+
+#### 2.4 Create NFS Server
+
+Modify the values.yaml in the nfs-provisioner with persistent volume name, disk name and node affinity zones and install nfs-provisioner using helm.
+
+```bash
+cd GKE
+
+helm install mynfs ./nfs-provisioner/
```

Comment:
@jagadeeshi2i Please mention that one needs to update the nfs-provisioner/values.yaml and change the <nfs-disk> to match the name that user used for creating